### PR TITLE
Update CDash config to ensure https is used for submit step

### PR DESCRIPTION
**Pull request by biddisco**

### CODE REVIEW

File: CTestConfig.cmake [hunk 1]
```diff
@@ -13,7 +13,12 @@
...
-set(CTEST_SUBMIT_URL
-    "https://cdash.cscs.ch/submit.php?project=${CTEST_PROJECT_NAME}"
-)
+set(CTEST_DROP_SITE "cdash.cscs.ch")
+set(CTEST_DROP_LOCATION "/submit.php?project=pika")
+set(CTEST_DROP_SITE_CDASH TRUE)
+
+#set(CTEST_SUBMIT_URL
+#    "https://monteggio-new.cscs.ch/submit.php?project=pika"
+#)
```
Comment (msimberg) [inline CTestConfig.cmake:24]: Was this for testing?

Comment (msimberg) [inline CTestConfig.cmake:24]:
```suggestion
```

Comment (biddisco) [reply CTestConfig.cmake]: Sorry. Thought I had deleted that. I have removed it and force pushed (yes it was for testing)

Comment (msimberg) [reply CTestConfig.cmake]: No worries, and thanks!

**Comment by msimberg** [comment]
tests cscs

**Comment by msimberg** [comment]
Assuming things are working correctly the jenkins jobs should soon come in with valid build ids...

File: CTestConfig.cmake [hunk 2]
```diff
@@ -13,7 +13,8 @@
 #
 set(CTEST_PROJECT_NAME "pika")
 set(CTEST_NIGHTLY_START_TIME "00:00:00 GMT")
+set(CTEST_DROP_METHOD "https")
 
-set(CTEST_SUBMIT_URL
-    "https://cdash.cscs.ch/submit.php?project=${CTEST_PROJECT_NAME}"
-)
+set(CTEST_DROP_SITE "cdash.cscs.ch")
+set(CTEST_DROP_LOCATION "/submit.php?project=pika")
```
Comment (aurianer) [inline CTestConfig.cmake:19]: should we still use `${CTEST_PROJECT_NAME}` here?

**Review by aurianer — APPROVED** [review]
LGTM thanks!

File: CTestConfig.cmake (hunk 2 above)
Comment (biddisco) [reply CTestConfig.cmake:19]: as long as `${CTEST_PROJECT_NAME}` expands to pika, then it doesn't really matter as far as the dashboard is concerned

**Comment by msimberg** [comment]
test cscs

**Comment by msimberg** [comment]
test cscs

**Comment by msimberg** [comment]
The CI itself isn't happy, but the CDash submission works, so I'm merging this. Thanks @biddisco.

bors merge

**Pull request merged**